federation/pkg/kubefed/init/init.go

```diff
@@ -485,6 +507,21 @@ func createAPIServer(clientset *client.Clientset, namespace, name, image, creden
 		return dep, nil
 	}
 
+	if !cloudEnvironment {
```

Comment:
totally agree with you. I was reluctant to introduce new flags. what you are bringing is more variety of use cases and thanks for the review.